synapse/lib/storm.py

```diff
@@ -4530,3 +4530,69 @@ async def execStormCmd(self, runt, genr):
                             break
 
                 yield node, path
+
+class RunAsCmd(Cmd):
+    '''
+    Execute a storm query as a specified user.
+
+    NOTE: This command requires admin privileges.
+
+    Examples:
+
+        // Create a node as another user.
+        runas someuser { [ inet:fqdn=foo.com ] }
+    '''
+
+    name = 'runas'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('user', help='The user name or iden to execute the storm query as.')
+        pars.add_argument('storm', help='The storm query to execute.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        if runt.user is not None and not runt.user.isAdmin():
```

Comment:
runt.user should not ever be None, maybe we should just deref it and blow up if it is?